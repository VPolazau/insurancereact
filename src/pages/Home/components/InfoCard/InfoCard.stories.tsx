import { InfoCard } from './InfoCard';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof InfoCard> = {
    title: 'UI/InfoCard',
    tags: ['autodocs'],
    component: InfoCard,
};

export default meta;
type Story = StoryObj<typeof InfoCard>;

export const Card: Story = {
    args: {
        children: <span style={{ fontSize: '50px' }}>Card</span>,
    },
};
