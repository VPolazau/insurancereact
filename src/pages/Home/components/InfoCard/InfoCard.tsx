import React from 'react';
import cn from 'classnames';

import styles from './styles.module.css';

export interface IInfoCard {
    className?: string;
    children: React.ReactElement;
}
export const InfoCard: React.FC<IInfoCard> = ({ className, children }) => {
    return (
        <div className={cn(styles.infoCardBasic, className)} data-testid="infoCard">
            <div className={styles.rectangle} />
            {children}
        </div>
    );
};
