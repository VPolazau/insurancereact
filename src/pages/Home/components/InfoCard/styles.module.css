.infoCardBasic {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 20px 19px;
    gap: 4px;
    width: 282px;
    border: 1px solid #22C062;
    border-radius: 8px;
}

.rectangle {
    width: 10px;
    height: 10px;
    background-color: #22C062;
    border-radius: 1px;
    transform: rotate(45deg);
    margin-bottom: 4px;
    transition: transform .5s ease-in-out, background-color .5s ease-in-out;
}

.infoCardBasic:hover .rectangle {
    background-color: #1F1D1D;
    transform: rotate(-45deg);
}

@media screen and (max-width: 1270px) {
    .infoCardBasic {
        padding: 19px;       
        flex: 0 0 calc(50% - 12px); /* Занимает всю доступную ширину без половина gap */
    } 
 }

@media screen and (max-width: 480px) {
    .infoCardBasic {
        padding: 20px;
        width: 100%;
    }
}