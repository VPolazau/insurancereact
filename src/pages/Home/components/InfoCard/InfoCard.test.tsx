import { render, screen } from '@testing-library/react';

import { InfoCard } from './InfoCard';

describe('InfoCard', () => {
    it('InfoCard with mock data', () => {
        render(<InfoCard className="mockClassInfoCard" children={<h5>Some mock text...</h5>} />);
        expect(screen.getByTestId('infoCard').textContent).toBe('Some mock text...');
        expect(screen.getByTestId('infoCard').classList.contains('mockClassInfoCard')).toBe(true);
        expect(screen.getByTestId('infoCard').classList.contains('infoCardBasic')).toBe(true);
        expect(screen.getByTestId('infoCard')).toBeInTheDocument();
        expect(screen.getByTestId('infoCard')).toMatchSnapshot();
    });
});
