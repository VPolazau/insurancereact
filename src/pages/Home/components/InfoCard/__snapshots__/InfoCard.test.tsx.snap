// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard InfoCard with mock data 1`] = `
<div
  class="infoCardBasic mockClassInfoCard"
  data-testid="infoCard"
>
  <div
    class="rectangle"
  />
  <h5>
    Some mock text...
  </h5>
</div>
`;
